opensrp-app/src/main/java/org/smartregister/repository/TaskRepository.java

```diff
@@ -406,4 +406,29 @@ public boolean updateTaskStructureIdFromStructure(List<Location> locations) {
         }
     }
 
+    /**
+     * This method updates the task.structure_id field with
+     * ids of existing structures where the structure._id equals
+     * the task.for value.
+     *
+     * <p>
+     * This is only done for tasks that have a null structure_id field
+     *
+     * @return bolean indicating whether the update was successful
+     */
+    public boolean updateTaskStructureIdsFromExistingStructures() {
+
+        try {
+            getReadableDatabase().execSQL(String.format("UPDATE %s SET %s =(SELECT %s FROM structure WHERE %s = %s) WHERE %s IS NULL",
+                    TASK_TABLE, STRUCTURE_ID, ID, ID, FOR, STRUCTURE_ID));
+
+            getReadableDatabase().execSQL(String.format("UPDATE %s SET %s =(SELECT %s FROM ec_family_member WHERE base_entity_id = %s) WHERE %s IS NULL",
```

Comment:
You should pass in the table that has the structureId i.e `ec_family_member` as a param to this method